cmd/infra/powervs/service_id.go

```diff
@@ -0,0 +1,153 @@
+	cco "github.com/openshift/cloud-credential-operator/pkg/cmd/provisioning/ibmcloud"
+	ccoibmcloud "github.com/openshift/cloud-credential-operator/pkg/ibmcloud"
+)
+
+var kubeCloudControllerManagerCR = `
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: openshift-powervs-cloud-controller-manager
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: IBMCloudPowerVSProviderSpec
+    policies:
+    - attributes:
+      - name: resourceType
+        value: resource-group
+      roles:
+      - crn:v1:bluemix:public:iam::::role:Viewer
+    - attributes:
+      - name: serviceName
+        value: is
+      roles:
+      - crn:v1:bluemix:public:iam::::role:Editor
+      - crn:v1:bluemix:public:iam::::role:Operator
+      - crn:v1:bluemix:public:iam::::role:Viewer
+    - attributes:
+      - name: serviceName
+        value: power-iaas
+      roles:
+      - crn:v1:bluemix:public:iam::::role:Viewer
+      - crn:v1:bluemix:public:iam::::serviceRole:Reader
+      - crn:v1:bluemix:public:iam::::serviceRole:Manager
+`
+var nodePoolManagementCR = `
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: openshift-capi-powervs
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: IBMCloudPowerVSProviderSpec
+    policies:
+    - attributes:
+      - name: serviceName
+        value: power-iaas
+      roles:
+      - crn:v1:bluemix:public:iam::::serviceRole:Manager
+      - crn:v1:bluemix:public:iam::::role:Editor
+`
+
+var ingressOperatorCR = `
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: openshift-ingress-powervs
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: IBMCloudPowerVSProviderSpec
+    policies:
+    - attributes:
+      - name: serviceName
+        value: internet-svcs
+      roles:
+      - crn:v1:bluemix:public:iam::::serviceRole:Manager
+      - crn:v1:bluemix:public:iam::::role:Editor
+`
+
+const (
+	KubeCloudControllerManager = "kubeCloudControllerManager"
+	NodePoolManagement         = "nodePoolManagement"
+	IngressOperator            = "ingressOperator"
+)
+
+var SecretsToManage = map[string]string{
+	KubeCloudControllerManager: kubeCloudControllerManagerCR,
+	NodePoolManagement:         nodePoolManagementCR,
+	IngressOperator:            ingressOperatorCR,
+}
+
+// createServiceIDClient creates cloud credential operator's serviceID client
+func createServiceIDClient(name, APIKey, accountID, resourceGroupID, crYaml, secretRefName, secretRefNamespace string) (*cco.ServiceID, error) {
```

Comment:
lets add some unit tests around this function.

Reply:
Added UT validating yaml decoding related err, since remaining involves actual network call and client created is not exporting any properties we passed, could not do assertion on those. 
Please take a look.